core/src/test/java/io/questdb/griffin/engine/groupby/TimestampSamplerFactoryTest.java

```diff
@@ -81,26 +57,83 @@ public void testNoQualifier() {
         assertFailure(100, "expected interval qualifier", "45", 98);
     }
 
+    @Test
+    public void testMinutes() throws NumericException, SqlException {
+        final StringSink sink = new StringSink();
+        final long ts = TimestampFormatUtils.parseUTCTimestamp("2022-04-23T10:33:00.123456Z");
+        for (int k = 0; k < 61; k++) {
+            final TimestampSampler sampler = createTimestampSampler(k, 'm', sink);
+            final long bucketSize = 60 * (k == 0 ? 1 : k);
+            final long expectedTs = ts - ts % (Timestamps.SECOND_MICROS * bucketSize);
+            for (int i = 0; i < bucketSize; i+=4) {
+                long actualTs = sampler.round(expectedTs + i * Timestamps.SECOND_MICROS);
+                if (expectedTs != actualTs) {
+                    Assert.fail(String.format(
+                            "Failed at: %s, i: %d. Expected: %s, actual: %s",
+                            sink, i, Timestamps.toString(expectedTs), Timestamps.toString(actualTs))
+                    );
+                }
+            }
+        }
+    }
+
     @Test
     public void testSeconds() throws NumericException, SqlException {
-        StringSink sink = new StringSink();
+        final StringSink sink = new StringSink();
+        final long ts = TimestampFormatUtils.parseUTCTimestamp("2022-04-23T10:33:00.123456Z");
         for (int k = 0; k < 61; k++) {
-            sink.clear();
-            if (k > 0) {
-                sink.put(k).put('s');
-            } else {
-                sink.put('s');
+            final TimestampSampler sampler = createTimestampSampler(k, 's', sink);
+            final long bucketSize = Timestamps.SECOND_MILLIS * (k == 0 ? 1 : k);
```

Comment:
same logic as stated in previous comment, should be `Timestamps.SECOND_MICROS * k...`

Reply:
So can I try like this? Thanks!
```
final long bucketSize = Timestamps.SECOND_MICROS * (k == 0 ? 1 : k);
final long expectedTs = ts - ts % bucketSize;
for (int i = 0; i < bucketSize; i+=4 * Timestamps.MILLI_MICROS) {
     long actualTs = sampler.round(expectedTs + I);
}
```

Reply:
I think that if you leave the essence of the test for seconds and micros as it was, except for the use of the `createTimestampSampler` function we will be on the win.

Reply:
Ok, thanks! I've pulled the newest version, please check it! Appreciate it! @marregui

Reply:
Hi, I've pulled the newest version, please check, thanks! @marregui